File: CWE_Python/src/CWE/CWE862/NonComplaint.py
class User:
    def __init__(self, user, resource):
        self.name = user
        self.permission = resource


def operation(user, resource):
    print("Unsafe Operation::", resource)


user = User("name",["read:file1", "write:file2"]);
operation(user, "write:file1");

File: CWE_Java/src/main/java/com/example/cwe/CWE862/NonComplaint/NonComplaint.java
package com.example.cwe.CWE862.NonComplaint;

import java.util.*;

//      CWE-862: Missing Authorization

public class NonComplaint {

    // Method that requires authorization
    public void sensitiveOperation(User user, String resource) {
        System.out.println("Unsafe Operation::"+resource);
    }

    // Example usage
    public static void main(String[] args) {
        User user = new User("user", Arrays.asList("read:file1", "write:file2"));
        NonComplaint service = new NonComplaint();
        service.sensitiveOperation(user, "write:file1");
    }
}




